frontend/src/components/system-upgrade/upgrade/index.vue

```diff
@@ -0,0 +1,127 @@
+                    {{ upgradeInfo.testVersion }}
+                </el-radio>
+            </el-radio-group>
+            <MdEditor v-model="upgradeInfo.releaseNote" previewOnly :theme="isDarkTheme ? 'dark' : 'light'" />
+        </div>
+        <template #footer>
+            <span class="dialog-footer">
+                <el-button @click="drawerVisible = false">{{ $t('commons.button.cancel') }}</el-button>
+                <el-button type="primary" @click="onUpgrade">{{ $t('setting.upgradeNow') }}</el-button>
+            </span>
+        </template>
+    </el-drawer>
+</template>
+
+<script setup lang="ts">
+import DrawerHeader from '@/components/drawer-header/index.vue';
+import { loadReleaseNotes, upgrade } from '@/api/modules/setting';
+import MdEditor from 'md-editor-v3';
+import i18n from '@/lang';
+import 'md-editor-v3/lib/style.css';
+import { MsgSuccess } from '@/utils/message';
+import { ref } from 'vue';
+import { GlobalStore } from '@/store';
+import { ElMessageBox } from 'element-plus';
+import { storeToRefs } from 'pinia';
+
+const globalStore = GlobalStore();
+const { isDarkTheme } = storeToRefs(globalStore);
+
+const drawerVisible = ref(false);
+const upgradeInfo = ref();
+const refresh = ref();
+const upgradeVersion = ref();
+
+interface DialogProps {
+    upgradeInfo: number;
+    upgradeVersion: string;
+}
+const acceptParams = (params: DialogProps): void => {
+    console.log(params);
+    upgradeInfo.value = params.upgradeInfo;
+    upgradeVersion.value = params.upgradeVersion;
+    drawerVisible.value = true;
+};
+
+const emit = defineEmits(['search']);
+
+const handleClose = () => {
+    drawerVisible.value = false;
+};
+
+const changeOption = async () => {
+    const res = await loadReleaseNotes(upgradeVersion.value);
+    upgradeInfo.value.releaseNote = res.data;
+};
+
+const onUpgrade = async () => {
+    ElMessageBox.confirm(i18n.global.t('setting.upgradeHelper', i18n.global.t('commons.button.upgrade')), {
+        confirmButtonText: i18n.global.t('commons.button.confirm'),
+        cancelButtonText: i18n.global.t('commons.button.cancel'),
+        type: 'info',
+    }).then(async () => {
+        globalStore.isLoading = true;
+        await upgrade(upgradeVersion.value);
+        globalStore.isOnRestart = true;
+        drawerVisible.value = false;
+        MsgSuccess(i18n.global.t('commons.msg.operationSuccess'));
+        emit('search');
+    });
+};
+
+defineExpose({
+    acceptParams,
+});
+</script>
+
+<style lang="scss" scoped>
+.line-height {
+    line-height: 25px;
+}
+.panel-MdEditor {
+    height: calc(100vh - 330px);
+    .tag {
+        margin-top: -6px;
+        margin-left: 20px;
+        vertical-align: middle;
+    }
+    :deep(.md-editor-preview) {
+        font-size: 14px;
+    }
+    :deep(.default-theme h2) {
+        color: var(--dark-gold-base-color);
+        margin: 13px, 0;
+        padding: 0;
+        font-size: 16px;
+    }
+}
+</style>
```

Comment:
The provided code has a few differences that you can optimize for better performance and readability:

**Differences:**
1. The `md-editor` was updated to utilize the latest version, which should be checked against the current versions of MD editor before using it.
2. The code is not properly formatted and could benefit from more meaningful variable names.

### Optimization Suggestions:
Since this seems like personal work, I cannot run tests directly as an AI. However, here's some general advice:

- **Format Code:** Use semicolons instead of spaces around each statement.
- **Use meaningful variables:** Instead of referring to `<el-drawer>` with its full name (`<el-drawer>`) every time, refer to it simply as `'...'<elm.drawer>'`.
- **Consistent Naming:** Ensure consistency across all parts of the application naming conventions throughout your project.
- **Function Names/Symbolic References:** Instead of relying on long functions such as `changeOption`, consider creating smaller, more specific functions and symbols if necessary.
- **Documentation:** Make sure there is documentation explaining what each block does.

Please review these optimizations and see how they impact the overall functionality of your component. If necessary, you might also need to refactor methods into their own functions to improve their clarity and maintainability.

Remember, improving readability and maintaining consistent standards are crucial steps towards making your codebase cleaner and easier to understand for both yourself and others who will read and use the code in the future.